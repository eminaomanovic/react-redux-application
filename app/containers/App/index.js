/**
 *
 * App
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 */

import React from 'react';
import styled from 'styled-components';
import 'bootstrap/dist/css/bootstrap.min.css';
import history from 'utils/history';

import HomePage from 'containers/HomePage/Loadable';
import DetailPage from 'containers/DetailPage/Loadable';
import { Route, Router, Switch } from 'react-router-dom';
import GlobalStyle from '../../global-styles';

const AppWrapper = styled.div`
  margin: 0 auto;
  display: flex;
  min-height: 100%;
  flex-direction: column;
`;

export default function App() {
  return (
    <AppWrapper>
      <Router history={history}>
        <Switch>
          <Route exact path="/" component={HomePage} history={history} />
          <Route
            path="/detailPage/:movieType/:videoId"
            component={DetailPage}
            history={history}
          />
        </Switch>
      </Router>
      <GlobalStyle />
    </AppWrapper>
  );
}
