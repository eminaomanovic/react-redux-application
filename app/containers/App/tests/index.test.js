import React from 'react';
import { Provider } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import configureStore from 'redux-mock-store';
import { ConnectedRouter } from 'connected-react-router/immutable';
import history from 'utils/history';
import App from '../index';
const mockStore = configureStore([]);

describe('<App />', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      loading: true,
      error: false,
    });
  });

  it('should render with videos', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <App />
        </ConnectedRouter>
      </Provider>,
    );
    const component = renderer.getRenderOutput();
    expect(component).toMatchSnapshot();
  });
});
