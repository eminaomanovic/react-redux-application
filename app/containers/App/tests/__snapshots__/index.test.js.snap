// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render with videos 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Memo(ConnectFunction)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App />
  </Memo(ConnectFunction)>
</Context.Provider>
`;
